mqr/tests/test_views.py

```diff
@@ -121,3 +122,120 @@ def test_faq_message(self, mock_get_message_details):
         )
 
         mock_get_message_details.assert_called_with("BCM_week_pre22_faq1")
+
+
+class StrataRandomization(APITestCase):
+    url = reverse("mqr_randomstrataarm")
+
+    def test_random_arm_unauthorized_user(self):
+        """
+        unauthorized user access denied
+        Returns: status code 401
+
+        """
+
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_random_arm(self):
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+
+        MqrStrata.objects.create(
+            province="EC",
+            weeks_pregnant_bucket="21-25",
+            age_bucket="31+",
+            next_index=3,
+            order="RCM_SMS,RCM_BCM,RCM,ARM,BCM",
+        )
+
+        response = MqrStrata.objects.get(
```

Comment:
I would suggest calling the api endpoint here, then asserting on the status code and response

Reply:
Maybe change this test so that there is no strata before calling the endpoint
check status code
query strata and compare body to the strata that is found